common/headtracker/head_tracker.go

```diff
@@ -250,8 +250,9 @@ func (ht *headTracker[HTH, S, ID, BLOCK_HASH]) handleNewHead(ctx context.Context
 		}
 	} else {
 		ht.log.Debugw("Got out of order head", "blockNum", head.BlockNumber(), "head", head.BlockHash(), "prevHead", prevHead.BlockNumber())
-		prevUnFinalizedHead := prevHead.BlockNumber() - int64(ht.config.FinalityDepth())
-		if head.BlockNumber() < prevUnFinalizedHead {
+		prevLatestFinalized := prevHead.LatestFinalizedHead()
+
+		if prevLatestFinalized != nil && prevLatestFinalized.IsValid() && head.BlockNumber() <= prevLatestFinalized.BlockNumber() {
```

Comment:
Add isValid check

Reply:
It is already checking `prevLatestFinalized.IsValid()`